traits/observers/parsing.py

```diff
@@ -0,0 +1,262 @@
+        The notify flag stack.
+
+    Returns
+    -------
+    expression : traits.observers.expression.Expression
+    """
+    expressions = (
+        _handle_tree(tree, default_notifies=default_notifies)
+        for tree in trees
+    )
+    return _expr_module.join_(*expressions)
+
+
+def _handle_parallel(trees, default_notifies):
+    """ Handle expressions joined in parallel using "," connectors.
+
+    Parameters
+    ----------
+    trees : list of lark.tree.Tree
+        The children tree for the "parallel" rule.
+        It should contain one or more items.
+    default_notifies : list of boolean
+        The notify flag stack.
+
+    Returns
+    -------
+    expression : traits.observers.expression.Expression
+    """
+    expressions = (
+        _handle_tree(tree, default_notifies=default_notifies) for tree in trees
+    )
+    return reduce(operator.or_, expressions)
+
+
+@contextlib.contextmanager
+def _notify_flag(default_notifies, value):
+    """ Context manager to push the notify to the given stack.
+    Upon exiting the context, pop the flag out of the stack.
+
+    Parameters
+    ----------
+    default_notifies : list of boolean
+        The notify flag stack.
+    value : boolean
+        Notify flag to push.
+    """
+    default_notifies.append(value)
+    try:
+        yield
+    finally:
+        notify = default_notifies.pop()
+        if notify is not value:
+            raise RuntimeError("Default notify flag unexpectedly changed.")
+
+
+def _handle_notify(trees, default_notifies):
+    """ Handle trees wrapped with the notify flag set to True,
+    indicated by the existence of "." suffix to an element.
+
+    Parameters
+    ----------
+    trees : list of lark.tree.Tree
+        The children tree for the "notify" rule.
+        It contains only one item.
+    default_notifies : list of boolean
+        The notify flag stack.
+
+    Returns
+    -------
+    expression : traits.observers.expression.Expression
+    """
+    with _notify_flag(default_notifies, True):
+        return _handle_last(trees, default_notifies=default_notifies)
+
+
+def _handle_quiet(trees, default_notifies):
+    """ Handle trees wrapped with the notify flag set to True,
+    indicated by the existence of ":" suffix to an element.
+
+    Parameters
+    ----------
+    trees : list of lark.tree.Tree
+        The children tree for the "quiet" rule.
+        It contains only one item.
+    default_notifies : list of boolean
+        The notify flag stack.
+
+    Returns
+    -------
+    expression : traits.observers.expression.Expression
+    """
+    with _notify_flag(default_notifies, False):
+        return _handle_last(trees, default_notifies=default_notifies)
+
+
+def _handle_last(trees, default_notifies):
+    """ Handle trees when the notify is not immediately specified
+    as a suffix. The last notify flag will be used.
+
+    e.g. In "a.[b,c]:.d", the element "b" should receive a notify flag
```

Comment:
Is the `:.` here intentional, or a typo?

Reply:
Is a typo. Thanks!